### ISSUE TITLE
[Bug]: Screenreader(Narrator) can't read the <Option>s inside a <Dialog>

### ISSUE THREAD

**Issue by lancelodliu** [microsoft/fluentui#27108]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
    Memory: 10.38 GB / 31.89 GB
  Browsers:
    Chrome: 110.0.5481.178
    Edge: Spartan (44.19041.1266.0), Chromium (110.0.1587.63)
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/lingering-microservice-0662kh?file=/example.tsx

## Actual Behavior
Screenreader doesn't read the option text.

## Expected Behavior
Screenreader reads the option text as "XXX, x of N".

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by lancelodliu** [comment]
Only happened in Edge + Windows 10.
For Chrome, Windows 10 and 11's narrator can read it out.
For Edge, Windows 11 works as expected but Windows 10 fails.

**Comment by lancelodliu** [comment]
Any luck on this item?
